k2/csrc/fsa_utils.cu

```diff
@@ -0,0 +1,238 @@
+/**
+ * @brief Utilities for reading, writing and creating FSAs.
+ *
+ * @copyright
+ * Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include <sstream>
+#include <vector>
+
+#include "k2/csrc/context.h"
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+
+// field separator within a line for a text form FSA
+static constexpr const char *kDelim = " \t";
+
+// Convert a string to an integer. Abort the program on failure.
+static int32_t StringToInt(const std::string &s) {
+  K2_CHECK_EQ(s.empty(), false);
+
+  bool ok = false;
+  char *p = nullptr;
+  // std::strtol requires a `long` type
+  long n = std::strtol(s.c_str(), &p, 10);  // NOLINT
```

Comment:
Maybe it will be easy to use `std::stoi` and convert the exception to FATAL.

Reply:
stoi will throw an exception on failure. I think we don't want exceptions in K2.